### PR TITLE
rootless: temporarily discard "kernel.domainname" sysctl

### PR DESCRIPTION
**Pull request by thaJeztah**
Trying to run a container with the `--domainname` set currently fails if rootless-mode is enabled.

This patch temporarily discards the "kernel.domainname" when running in rootless mode, as a workaround for docker/for-linux#743.

This workaround should be removed once opencontainers/runc#2091 is resolved.

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: rootless/specconv/specconv_linux.go [hunk 1]
```diff
@@ -67,6 +69,14 @@ func toRootless(spec *specs.Spec, v2Controllers []string, currentOOMScoreAdj int
 		}
 	}
 
+	// FIXME: Temporary workaround for https://github.com/docker/for-linux/issues/743.
+	//        Remove this once https://github.com/opencontainers/runc/issues/2091 is
+	//        resolved in runc.
+	if _, ok := spec.Linux.Sysctl["kernel.domainname"]; ok {
+		logrus.Info("Domainname is currently not supported when running in rootless mode. Discarding 'kernel.domainname' sysctl. See https://github.com/docker/for-linux/issues/743 for details.")
+		delete(spec.Linux.Sysctl, "kernel.domainname")
+	}
```
Comment (thaJeztah) [inline rootless/specconv/specconv_linux.go:78]: Was in doubt what would be better; `discard` the option (and log, to make it slightly visible), or to return an informative error here?

@AkihiroSuda @cpuguy83 WDYT?

**Comment by thaJeztah** [comment]
Ah... complete brain-fart; this is for "userns" mode, not rootless

**Comment by raulgomis** [comment]
@thaJeztah are you planning to integrate that change for userns remap? It's currently blocking some users to migrate to v19 in Bitbucket Pipelines.

**Comment by jonasgeiler** [comment]
> @thaJeztah wrote:
> Ah... complete brain-fart; this is for "userns" mode, not rootless

If I understand correctly you closed this patch since docker/for-linux#743 is about userns and not docker rootless-mode, right?
Well, when I fully set up docker in rootless-mode right now I still get the error message anytime I try to use `--domainname` (or `domainname:` in compose):
```
$ docker run -it --rm --domainname example.com alpine
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: open /proc/sys/kernel/domainname: permission denied: unknown.
```

I think it would be good to pick up this issue again. It would be enough if docker at least threw a more comprehensible error message, like proposed in this PR.
The linked issue for runc (opencontainers/runc#2091) hasn't been touched for almost 5 years now too...

**Pull request closed without merging**